content_server/openttd/tcp_content.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+import logging
+
+from .protocol.exceptions import PacketInvalid
+from .protocol.source import Source
+from .receive import OpenTTDProtocolReceive
+from .send import OpenTTDProtocolSend
+
+log = logging.getLogger(__name__)
+
+
+class OpenTTDProtocolTCPContent(asyncio.Protocol, OpenTTDProtocolReceive, OpenTTDProtocolSend):
+    def __init__(self, callback_class):
+        super().__init__()
+        self._callback = callback_class
+        self._callback.protocol = self
+        self._queue = asyncio.Queue()
+        self._data = b""
+        self.is_ipv6 = None
+
+        asyncio.create_task(self._process_queue())
+
+    def connection_made(self, transport):
+        self.transport = transport
+        socket_addr = transport.get_extra_info("peername")
+        self.source = Source(self, socket_addr, socket_addr[0], socket_addr[1])
+        # TODO -- Support PROXY protocol
```

Comment:
what is the "PROXY protocol" ?

Reply:
HAProxy invention that is awesome:
https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-proxy-protocol.html

I most likely have to support it, so we know real IPs here, instead of the internal AWS LoadBalancer :D